subiquity/models/filesystem.py

```diff
@@ -1638,31 +1644,38 @@ def match_install_media(disk):
             matchers.append(match_devpath)
         if "ssd" in match:
             matchers.append(match_ssd)
+        if "size" in match or "ssd" in match:
+            matchers.append(match_not_in_use)
 
         return matchers
 
-    def disk_for_match(self, disks, match):
-        log.info(f"considering {disks} for {match}")
-        matchers = self._make_matchers(match)
-        candidates = []
-        for candidate in disks:
-            if candidate.size == 0:
-                continue
-            for matcher in matchers:
-                if not matcher(candidate):
-                    break
-            else:
-                candidates.append(candidate)
-        if "size" in match or "ssd" in match:
-            candidates = [c for c in candidates if not c._has_in_use_partition]
+    def _sorted_matches(self, disks: Sequence[_Device], match: dict):
         if match.get("size") == "smallest":
-            candidates.sort(key=lambda d: d.size)
-        if match.get("size") == "largest":
-            candidates.sort(key=lambda d: d.size, reverse=True)
-        if candidates:
-            log.info(f"For match {match}, using the first candidate from {candidates}")
-            return candidates[0]
-        log.info(f"For match {match}, no devices match")
+            disks.sort(key=lambda d: d.size)
+        elif match.get("size") == "largest":
+            disks.sort(key=lambda d: d.size, reverse=True)
+        return disks
+
+    def _filtered_matches(self, disks: Sequence[_Device], match: dict):
+        matchers = self._make_matchers(match)
+        return [disk for disk in disks if all(match_fn(disk) for match_fn in matchers)]
+
+    def disk_for_match(
```

Comment:
Nothing really to do with your change but do any of the methods related to this actually refer to self? Maybe we can move them to a helper module at some point.